### PR TITLE
Plugin script: Fix ES_HOME with spaces

### LINKED ISSUE #12504: Space in directory structure fails plugin binary
When one of the parent directories has a space in there the plugin binary can't find the correct path.

```
$ pwd
/home/richard/temp dir/elasticsearch-1.7.0
$ ./bin/plugin
Error: Could not find or load main class dir.elasticsearch-1.7.0.config
```

### PR DESCRIPTION
**Pull request by nik9000**
When ES_HOME has spaces we get funky "cannot find main class <part of a
directory>" errors. This makes them stop by escaping directories and using
eval instead of exec.

Closes #12504

### CODE REVIEW

**Comment by nik9000** [comment]
My bash foo is pretty weak but this makes the problem stop on my laptop. Feel free to tell me its silly and propose another fix.

File: distribution/src/main/resources/bin/plugin [hunk 1]
```diff
@@ -23,7 +23,7 @@ ES_HOME=`cd "$ES_HOME"; pwd`
 
 # Sets the default values for elasticsearch variables used in this script
 if [ -z "$CONF_DIR" ]; then
-  CONF_DIR="${packaging.plugin.default.config.dir}"
+  CONF_DIR="$ES_HOME/config"
```
Comment (tlrx) [inline distribution/src/main/resources/bin/plugin]: This should not be changed, the value `${packaging.plugin.default.config.dir}` is replaced at build time this line allows us to define the default value when `CONF_DIR` is not explicitly defined in the current env.

File: distribution/src/main/resources/bin/plugin [hunk 2]
```diff
@@ -31,12 +31,12 @@ if [ -z "$CONF_DIR" ]; then
 fi
 
 if [ -z "$CONF_FILE" ]; then
-  CONF_FILE="${packaging.plugin.default.config.file}"
+  CONF_FILE="$ES_HOME/config/elasticsearch.yml"
```
Comment (tlrx) [inline distribution/src/main/resources/bin/plugin]: Same as precedent comment

File: distribution/src/main/resources/bin/plugin [hunk 3]
```diff
@@ -31,12 +31,12 @@ if [ -z "$CONF_DIR" ]; then
...
 fi
 
 # The default env file is defined at building/packaging time.
-# For a ${packaging.type} package, the value is "${packaging.env.file}".
-ES_ENV_FILE="${packaging.env.file}"
+# For a tar.gz package, the value is "".
+ES_ENV_FILE=""
```
Comment (tlrx) [inline distribution/src/main/resources/bin/plugin]: Same as precedent comment

File: distribution/src/main/resources/bin/plugin [hunk 4]
```diff
@@ -98,11 +98,11 @@ if [ -e "$CONF_FILE" ]; then
...
-      properties="$properties -Des.default.config=$CONF_FILE"
+      properties="$properties -Des.default.config=\"$CONF_FILE\""
     ;;
   esac
 fi
 
 export HOSTNAME=`hostname -s`
 
-exec "$JAVA" $JAVA_OPTS $ES_JAVA_OPTS -Xmx64m -Xms16m -Delasticsearch -Des.path.home="$ES_HOME" $properties -cp "$ES_HOME/lib/*" org.elasticsearch.plugins.PluginManagerCliParser $args
+eval "$JAVA" $JAVA_OPTS $ES_JAVA_OPTS -Xmx64m -Xms16m -Delasticsearch -Des.path.home=\""$ES_HOME"\" $properties -cp \""$ES_HOME/lib/*"\" org.elasticsearch.plugins.PluginManager $args
```
Comment (tlrx) [inline distribution/src/main/resources/bin/plugin]: Why did you change the class name?

Comment (nik9000) [inline distribution/src/main/resources/bin/plugin]: Bleh - looks like I blindly applied this to the wrong branch. Ignore. Its garbage.

**Comment by tlrx** [comment]
@nik9000 I left some comments. Also, that would be nice if we can have an integration test for this (at least a *.bats test for now and maybe later something in `integration-tests.xml`?)

**Comment by nik9000** [comment]
Ignore this for now. I blindly merged it into the wrong branch. I know it applies to 1.7.0 so let me redo it there.

**Comment by nik9000** [comment]
Abandoned.

**Comment by nik9000** [comment]
Created against the proper path as #12508.
